aws-cs-lambda/pulumi/Program.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Pulumi;
+using Pulumi.Aws.Lambda;
+using Pulumi.Aws.Iam;
+
+class LambdaUtil
+{
+    public static Pulumi.AssetArchive buildArchive(string lambdaArchivePath)
+    {
+        var immutableDictBuilder = System.Collections.Immutable.ImmutableDictionary.CreateBuilder<string, AssetOrArchive>();
+        immutableDictBuilder.Add(".", new FileArchive(lambdaArchivePath));
+        return new Pulumi.AssetArchive(immutableDictBuilder.ToImmutable());
+    }
+    public static Role createLambdaRole()
+    {
+        var lambdaRole = new Role("lambdaRole", new RoleArgs
+        {
+            AssumeRolePolicy =
+            @"{
+                ""Version"": ""2012-10-17"",
+                ""Statement"": [
+                    {
+                        ""Action"": ""sts:AssumeRole"",
+                        ""Principal"": {
+                            ""Service"": ""lambda.amazonaws.com""
+                        },
+                        ""Effect"": ""Allow"",
+                        ""Sid"": """"
+                    }
+                ]
+            }",
+        });
+
+        var logPolicy = new RolePolicy("lambdaLogPolicy", new RolePolicyArgs
+        {
+            Role = lambdaRole.Id,
+            Policy =
+            @"{
+                ""Version"": ""2012-10-17"",
+                ""Statement"": [{
+                    ""Effect"": ""Allow"",
+                    ""Action"": [
+                        ""logs:CreateLogGroup"",
+                        ""logs:CreateLogStream"",
+                        ""logs:PutLogEvents""
+                    ],
+                    ""Resource"": ""arn:aws:logs:*:*:*""
+                }]
+            }"
+        });
+
+        return lambdaRole;
+    }
+}
+
+class Program
```

Comment:
I would put the `Program` class on top so that people would see it first. Usual C# practice is to have one class per file, but that's not required here. You could also merge two classes into one with two methods.